src/test_node.js

```diff
@@ -0,0 +1,124 @@
+// @ts-check
+var _ = require('lodash');
+
+class TestNode {
+  /**
+   * @param {import('./factory').Factory} factory
+   * @param {string} nodeName
+   */
+  constructor(factory, nodeName) {
+    this._factory = factory;
+    this._nodeName = nodeName;
+    this._attributes = {};
+    this._properties = [];
+    this._children = [];
+  }
+
+  /**
+   * @param {string} name
+   * @param {string} value
+   * @returns {TestNode}
+   * @chainable
+   */
```

Comment:
The type of `value` might be stricter than before, right? But requiring strings makes sense since we don't do any type conversions.

Reply:
Yea, technically the `type` is stricter as all the types used to be `any`. It won't break any existing code as you said we don't convert/check the values but it will throw TS errors on upgrading if the user is passing a number/boolean.